syncmymoodle/__main__.py

```diff
@@ -614,12 +614,34 @@ def sync(self):
                                     single=True,
                                 )
                             else:
-                                self.scanForLinks(
-                                    module.get("description", ""),
-                                    section_node,
-                                    course_id,
-                                    module_title=module["name"],
+                                # "Interactive" h5p videos
+                                if module["modname"] == "h5pactivity":
```

Comment:
Sorry, I went over this again: I think I finally figured out what felt off about this code despite it 'working in theory'; I think the control statements could a bit better. I understand why you would use `single=True` for a link here, but I think it would also be better if you put the `h5pactivity`-related functionality in the same place where other such links are handled -- like in the part which you later "excluded" (`if single and module_title != "h5pactivity":`), instead of creating two separate places to handle different types content (e.g. YouTube and H5P) belonging under the same set ("single" URLs).

There's a chance that I'm enforcing a view that's excessively subjective or potentially misinformed, as I'm still working my way around the codebase. Feel more than free to tell me so directly if you happen to disagree with this review.

Reply:
Hey @Romern, I am currently having some trouble with assessing how to move forward, especially because of already acknowledged problems (namely, #98) and if the difficulty in fixing that said issue if I proceed with merging this will be increased.

I am namely confused as to whether the input has to be sanitized first or sanitized within the function `scanForLinks` itself, as the latter already *somewhat* processes the input like that in the context of YouTube videos and because I'm also afraid of an edge case where scanning for links recursively may end up badly if, suddenly, a `h5pactivity` appears out of nowhere.

Reply:
> The first if should be combined with the else to an elif.

I mentioned that initially, thought I'd have to avoid that because "it would solve itself" if we were to both agree that maybe the entire functionality would go someplace else.

Reply:
> Also the usage of single=True does not make any sense as it is essentially ignored by the added condition in the method as you already noticed.

I did that because of [this](https://github.com/Romern/syncMyMoodle/pull/98/commits/3631bf2824442da25a94656b965bf90b6c749b9d#diff-a13b044f1b3c91e7aa9379a8cd47c7f412ef6bab145385853b16934053af4ffeL995) if-condition. The else-case somehow only searches for embedded links, thats why I want to go into the if-case. Thats why I think this if-condition is redundant, because the if-case has the same function.

Reply:
https://github.com/Romern/syncMyMoodle/blob/f8f10da7cf6f79cf25711dd690e756938c3ebf77/syncmymoodle/__main__.py#L973

Reply:
The whole `scanForLinks` should definitely be split into two functions, however, I do not consider this a blocker for this PR